### PR TITLE
READY: 'Back to You' button

### PR DESCRIPTION
**Pull request by TimSpeelman**
Solves #219

### CODE REVIEW

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 1]
```diff
@@ -47,6 +59,8 @@ function update(graph) {
 
     console.log(graph);
 
+    state.user_pk = state.user_pk ? state.user_pk : graph.focus_pk;
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/main.js]: This code can be replaced by `state.user_pk = state.user_pk || graph.focus_pk`.

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 2]
```diff
@@ -110,16 +124,27 @@ function handle_node_click(public_key) {
...
+}
+
+function onStep(public_key, neighbor_level){
+    state.request_pending = true;
 
-            // Store the previous focus node and its angle
-            var lk = state.data.local_keys.indexOf(public_key);
-            var newNode = state.data.nodes[lk];
-            var newTreeNode = find(state.tree.nodes, 'graphNode', newNode);
+    // First request?
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/main.js]: A comment with a question mark?

File: TriblerGUI/widgets/trustpage/js/radial_view/radial_simulation.js [hunk 3]
```diff
@@ -14,10 +14,10 @@ function RadialSimulation(options) {
     var defaults = {
         center_x: 0,
         center_y: 0,
-        alpha_decay: 0,
-        radial_force_strength: 0.2,
+        alpha_decay: 0.01,
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/radial_view/radial_simulation.js]: Why are these constants changed?

File: TriblerGUI/widgets/trustpage/js/request_data.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ function make_cors_request(method, url) {
  * @param callback The callback which is called with the GraphResponseData
  */
 function get_node_info(public_key, neighbor_level, callback) {
-    var url = "http://localhost:8085/multichain/network?focus_node=" + public_key + "&neighbor_level=" + neighbor_level;
+    var url = "http://localhost:8085/multichain/network?dataset=random&focus_node=" + public_key + "&neighbor_level=" + neighbor_level;
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/request_data.js]: Remove debug functionality.

File: TriblerGUI/widgets/trustpage/js/radial_view/radial_simulation.js (hunk 3 above)
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/radial_view/radial_simulation.js]: The animation very slowly dies out (when everything is already positioned correctly) so d3 stops calculating positions. This improves performance.

File: TriblerGUI/widgets/trustpage/index.html [hunk 5]
```diff
@@ -8,6 +8,7 @@
     <link rel="stylesheet" type="text/css" href="style.css">
 </head>
 <body>
+<button id="locate-me-button" onclick="locateMe()">Back To You</button>
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/index.html]: Please be consistent with the naming, also use back-to-you in id, function name etc.

File: TriblerGUI/widgets/trustpage/index.html [hunk 6]
```diff
@@ -34,6 +35,8 @@ <h2 class="title">
 
 <!-- MAIN -->
 <script src="js/style_config.js"></script>
+<script src="js/radial_view/radial_navigation.js"></script>
+<script src="js/radial_view/radial_stepping_animation.js"></script>
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/index.html]: Shouldn't this be `radial_stepping_animator`, to be consistent with the naming below?

File: TriblerGUI/widgets/trustpage/index.html [hunk 7]
```diff
@@ -34,6 +35,8 @@ <h2 class="title">
 
 <!-- MAIN -->
 <script src="js/style_config.js"></script>
+<script src="js/radial_view/radial_navigation.js"></script>
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/index.html]: `radial_navigator`

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 8]
```diff
@@ -24,11 +32,15 @@ var state = {
     previous_angle: 0,
     previous_focus_pk: null,
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/main.js]: You could use the shorthand syntax: `previous_focus_pk = focus_node = user_pk = null`

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 9]
```diff
@@ -47,6 +59,8 @@ function update(graph) {
 
     console.log(graph);
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/main.js]: Do we still need this debugging functionality here? If yes please add a string with explanation indicating what is logged.

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 10]
```diff
@@ -110,16 +124,27 @@ function handle_node_click(public_key) {
         console.log("Request pending, ignore new request");
     } else {
         if (public_key !== state.focus_pk) {
-            state.request_pending = true;
+            navigation.step(public_key);
+        }
+    }
+}
+
+function onStep(public_key, neighbor_level){
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/main.js]: This function feels better if he has proper documentation, so developers can read details about him.

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 11]
```diff
@@ -110,16 +124,27 @@ function handle_node_click(public_key) {
...
+    state.request_pending = true;
 
-            // Store the previous focus node and its angle
-            var lk = state.data.local_keys.indexOf(public_key);
-            var newNode = state.data.nodes[lk];
-            var newTreeNode = find(state.tree.nodes, 'graphNode', newNode);
+    // Subsequent requests
+    if(state.data) {
+        // Store the previous focus node and its angle
+        var lk = state.data.local_keys.indexOf(public_key);
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/main.js]: `lk` is a bad variable name

File: TriblerGUI/widgets/trustpage/js/main.js [hunk 12]
```diff
@@ -110,16 +124,27 @@ function handle_node_click(public_key) {
...
+        var newTreeNode = find(state.tree.nodes, 'graphNode', newNode);
+        if(newTreeNode) {
             state.previous_focus_pk = state.focus_pk;
             state.previous_angle = newTreeNode.alpha;
-
-            get_node_info(public_key, config.neighbor_level, onNewData)
         }
     }
 }
+
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/main.js]: documentation

File: TriblerGUI/widgets/trustpage/style.css [hunk 13]
```diff
@@ -130,3 +130,28 @@ svg {
...
+    padding: 5px;
+    background: transparent;
+    color: white;
+    font-weight: bold;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    margin-left: -500px;
+    -webkit-transition: all .3s;
+
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/style.css]: remove space

File: TriblerGUI/widgets/trustpage/js/radial_view/radial_node.js [hunk 14]
```diff
@@ -145,6 +145,56 @@ function RadialNodes(svg, options) {
...
+     * @param {String[]} keys - the public keys of the nodes to be unmarked
+     */
+    self.removeMarker = function (keys) {
+        self.selectAll().each(function (node) {
+            if (keys.indexOf(node.public_key) < 0) return;
+
+            d3.select(this).select('.marker')
+                .transition()
+                .duration(500)
+                .attr("r", "60")
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/radial_view/radial_node.js]: Move to config

File: TriblerGUI/widgets/trustpage/js/radial_view/radial_node.js [hunk 15]
```diff
@@ -145,6 +145,56 @@ function RadialNodes(svg, options) {
...
+                .attr("overflow", "visible")
+
+                .append("circle")
+                .attr("fill", "transparent")
+                .attr("stroke", function (node) { return node.is_user ? color_range[2] : color_range[0]; })
+                .attr("stroke-width", 2)
+
+                // Fade-in and reduce radius
+                .style("opacity", 0)
+                .attr("r", 60)
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/radial_view/radial_node.js]: Move to config

File: TriblerGUI/widgets/trustpage/index.html (hunk 6 above)
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/index.html]: No, only inspector is called like that.

File: TriblerGUI/widgets/trustpage/js/main.js (hunk 8 above)
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/main.js]: nope, not possible in js objects, only in var statements

File: TriblerGUI/widgets/trustpage/js/main.js (hunk 9 above)
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/main.js]: yes, lets remove this in the final pr

**Pull request merged**